tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -25,20 +25,39 @@
 from tests_e2e.tests.lib.agent_test import AgentTest
 
 
+class TestInfo(object):
+    """
+    Description of a test
+    """
+    # The class that implements the test
+    test_class: Type[AgentTest]
+    # If True, an error in the test blocks the execution of the test suite (defaults to False)
+    blocks_suite: bool
+
+    @property
+    def name(self) -> str:
+        return self.test_class.__name__
+
+    def __str__(self):
+        return self.name
+
+
 class TestSuiteInfo(object):
     """
     Description of a test suite
     """
     # The name of the test suite
     name: str
     # The tests that comprise the suite
-    tests: List[Type[AgentTest]]
+    tests: List[TestInfo]
     # Images or image sets (as defined in images.yml) on which the suite must run.
     images: List[str]
     # The location (region) on which the suite must run; if empty, the suite can run on any location
     location: str
     # Whether this suite must run on its own test VM
     owns_vm: bool
+    # Customization for the ARM template used when creating the test VM
+    template: str
```

Comment:
Added a new property, 'template', to allow test to use custom ARM templates to create the test VMs